src/adams_methods.jl

```diff
@@ -0,0 +1,655 @@
+        dy[i+1] = F(t[i+1], y[i+1])
+    end
+    return t, y
+end
+
+ode_am5(F::Function,y0, t)= ode_am(F::Function,y0, t,5)
+
+
+################################################################################
+# Variable Step Variable Order (VSVO) Adams-Moulton Solver for nonstiff problems
+#
+#     [T,X] = ODE113(ODEFUN, Y0, TSPAN,ORDER=4;kwords...) with TSPAN = [T0:TFINAL]
+#     or[T0,T1,...,TFINAL] for dense output
+#
+#     (Main ref: Hairer & Wanner 1996, Vol I, p.356-360, p.398-400, p.421-423)
+################################################################################
+function ode113(F, y0, tspan, order::Integer=4; kwords...)
+    # For y0 which don't support indexing.
+    F_ = (t, y) -> [F(t, y[1])]
+    t,y = ode113(F_, [y0], tspan, order; kwords...)
+    return t, vcat_nosplat(y)
+end
+
+"a variable stepsize, variable order multi-step solver for nonstiff ODEs"
+function ode113(F, y0::AbstractVector, tspan, order::Integer=4;adaptive_order = true,
+                                                      max_order = 13,
+                                                      reltol = 1.0e-9,
+                                                      abstol = 1.0e-9,
+                                                      minstep=abs(tspan[1] - tspan[end])/1e18,
+                                                      maxstep=abs(tspan[end] - tspan[1])/2.5,
+                                                      initstep=0.,
+                                                      facmax = 1.5,
+                                                      facmin = .1,
+                                                      fac = .8,
+                                                      norm=Base.norm,
+                                                      points =:all)
+
+    #-##########################################################################
+    # Initializations
+    #-##########################################################################
+    Et, Eyf, Ty = make_consistent_types(F, y0, tspan)
+    dof = length(y0)
+
+    # initialization of time span, which will have variable length
+    tspan = convert(Vector{Et}, tspan)
+    tstart = tspan[1]
+    tfinal = tspan[end]
+    t = [tstart]
+
+    # initialzation of y[n], which will have variable length as well
+    y = Array(Ty, 1)
+    allocate!(y, y0, dof)
+    # TODO: fix after https://github.com/JuliaLang/julia/issues/16667
+    # y[1] = deepcopy(y0)
+    y[1] = copy(y0)
+
+    # initialzation of dy[n], which will have variable length as well
+    Tdy = typeof(F(t[1], y[1]))
+    dy = Tdy[]
+    push!(dy,F(t[1], y[1]))
+
+    # preallocate work variables for t_np1, y_np1 and dy_np1  which are for
+    # for t_{n+1}, y(t_{n+1}), and dy(t_{n+1})
+    t_np1   = tspan[1]
+    y_np1   = similar(y0, Eyf, dof)
+    dy_np1  = similar(dy[1])
+
+    # variables for dense output
+
+    tout = [tstart]
+    iter = 2
+    yout = Array(Ty, 1)
+    allocate!(yout, y0, dof)
```

Comment:
This is pointless.  First, `allocate!` only makes sense to use with large vectors, not just length 1.  Then in line 199 you assign a copy of y0 to this slot, so the allocation is never used anyway.  199 would need to read `yout[1][:] = copy(y0)` or better `copy!(yout[1],y0)`.  However, these shenanigans will have no performance impact, instead write `yout = [copy(y0)]`.